### PR TITLE
Fix issue 2205

### LINKED ISSUE #2205: Chaining multiple finetuning jobs
### Describe the task. It can be a feature, documentation, etc.

When finetuning you need to set `istep: 0` but this also means when chaining jobs it is reset everytime, because for finetuning you need to have it be 0, so the learning rate can restart. This doesn't happen in pretraining, because you only set it base-config and not config.

### Hedgedoc URL, if you are keeping notes, plots, logs in hedgedoc.

_No response_

### Area

- [ ] datasets, data readers, data preparation and transfer
- [ ] model
- [ ] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation
- [ ] performance

### PR DESCRIPTION
**Pull request by sophie-xhonneux**
## Issue Number

Closes #2205 #2222 

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [x] I have performed a self-review of my code
-   [x] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

File: packages/common/src/weathergen/common/config.py [hunk 1]
```diff
@@ -392,8 +417,16 @@ def load_merge_configs(
         from_run_id = get_run_id_from_config(base_config)
     with open_dict(base_config):
         base_config.from_run_id = from_run_id
+
+    # In the case where we chain several finetuning jobs we don't want to reset the istep
+    # This case is detected by from_run_id being equal to run_id
+    istep = None
```
Comment (clessig) [inline packages/common/src/weathergen/common/config.py]: I disagree that it should be fixed at the level of the model code. The fix should be in launch-slurm where config should be passed to the first job and not subsequent ones.

**Review by grassesi — CHANGES_REQUESTED** [review]
How is this PR related to #2205? The code relevant to this issue contained in this PR was merged with #2206.
- Can #2205 be marked as closed?
- If this should be merged as an update to #2206 (for the compatibility code contained here), please remove the duplicated changes.

**Review by grassesi — APPROVED** [review]
I have a small patch [here](https://github.com/ecmwf/WeatherGenerator/pull/2228) that should address in conjunction with this PR the issue #2222 . Please take a look and feel free to merge.
